types/applications/sidebarTabs/sidebarDirectory.d.ts

```diff
@@ -0,0 +1,262 @@
+ */
+declare abstract class SidebarDirectory extends SidebarTab {
+  constructor(options?: Partial<Application.Options>);
+
+  /**
+   * References to the set of Entities which are displayed in the Sidebar
+   */
+  entities: Entity[];
+
+  /**
+   * Reference the set of Folders which exist in this Sidebar
+   * @defaultValue `null`
+   */
+  folders: Folder[];
+
+  /**
+   * @override
+   */
+  static get defaultOptions(): SidebarDirectory.Options;
+
+  /**
+   * The named entity which this Sidebar Directory contains
+   * @remarks This method is abstract in SidebarTab.
+   */
+  static get entity(): string;
+
+  /**
+   * The Entity collection which this Sidebar Directory contains
+   * @remarks This method is abstract in SidebarTab.
+   */
+  static get collection(): EntityCollection;
+
+  /**
+   * A reference to the Entity class which is displayed within this EntityCollection
+   */
+  static get cls(): ConstructorOf<Entity>;
+
+  /**
+   * Initialize the content of the directory by categorizing folders and entities into a hierarchical tree structure.
+   */
+  initialize(): void;
+
+  tree: SidebarDirectory.Tree;
+
+  /**
+   * Given an entity type and a list of entities, set up the folder tree for that entity
+   * @param folders  - The Array of Folder objects to organize
+   * @param entities - The Array of Entity objects to organize
+   * @returns A tree structure containing the folders and entities
+   */
+  static setupFolders(folders: Folder[], entities: Entity[]): SidebarDirectory.Tree;
+
+  /**
+   * Populate a single folder with child folders and content
+   * This method is called recursively when building the folder tree
+   */
+  protected static _populate(
+    folder: Folder,
+    folders: Folder[],
+    entities: Entity[],
+    {
+      allowChildren
+    }: {
+      /**
+       * @defaultValue `true`
+       */
+      allowChildren: boolean;
+    }
+  ): [Folder[], Entity[]];
+
+  /**
+   * @override
+   */
+  render(force?: boolean, context?: Partial<SidebarDirectory.RenderContext>): this | void;
+
+  /**
+   * @param options - (unused)
+   * @override
+   */
+  getData(options?: Application.RenderOptions): SidebarDirectory.Data | PlaylistDirectory.Data;
+
+  /**
+   * @param event - (unused)
+   * @override
+   */
+  protected _onSearchFilter(event: KeyboardEvent, query: string, html: HTMLElement): void;
+
+  /**
+   * Collapse all subfolders in this directory
+   */
+  collapseAll(): void;
+
+  /**
+   * Activate event listeners triggered within the Actor Directory HTML
+   */
+  activateListeners(html: JQuery): void;
+
+  /**
+   * Handle clicking on an Entity name in the Sidebar directory
+   * @param event - The originating click event
+   */
+  protected _onClickEntityName(event: JQuery.ClickEvent): void;
+
+  /**
+   * Handle new creation request
+   * @param event - The originating button click event
+   */
+  protected _onCreateEntity(event: JQuery.ClickEvent): Promise<Entity>; // TODO
```

Comment:
What's left to do here? Seems right to me. In theory, you could make `SidebarDirectory` generic in the `Entity` class and then return the correct `Entity` subclass here. Also applies to other places though.